executor/compiler.go

```diff
@@ -38,20 +38,29 @@ func (c *Compiler) Compile(ctx context.Context, node ast.StmtNode) (ast.Statemen
 	if err := plan.Validate(node, false); err != nil {
 		return nil, errors.Trace(err)
 	}
+
+	cacheable := plan.Cacheable(node)
+
 	p, err := plan.Optimize(ctx, node, is)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 
+	_, isDual := p.(*plan.TableDual)
+	if isDual {
+		cacheable = false
```

Comment:
Why dual cannot be cached?

Reply:
```
create table t(a bigint);
select * from t where exists (select * from t);
```

this query will be optimized to a `TableDual`